src/solver/writer/zip_writer.cpp

```diff
@@ -34,6 +35,12 @@ ZipWriteJob<ContentT>::ZipWriteJob(ZipWriter& writer,
 template<class ContentT>
 void ZipWriteJob<ContentT>::onExecute()
 {
+    // Don't write data if finalize() has been called
+    if (pState != ZipState::can_receive_data)
```

Comment:
In case of ctrl + c, if the goal here is let the started jobs finish but prevent others to start, 
maybe the block `if (pState != ZipState::can_receive_data) {...}` is not necessary, as we have such block in the **ZipWriter::addJob(...)** ?

Reply:
ok